src/DatadirTestsFromDirectoryProvider.php

```diff
@@ -86,6 +87,10 @@ protected function processOneTest(SplFileInfo $testSuite): void
             }
         }
 
+        if (file_exists($expectedStderrFile)) {
+            $expectedStderr = trim(file_get_contents($expectedStderrFile));
```

Comment:
trim pro vstupy jako takový je super 👍 , jen je otázka, jestli by měl být tady. Co když bude výstup `Test\nThis is an error\n`?

Reply:
Nad tím sem přemýšlel, sere mě jedna věc, když uložím ten `expected-stderr` tak ve stormu je defaultně zapnuté `Ensure line feed at file end on Save` to typicky chceš v `.php`, ale zrovna když vytváříš `expected-stderr` tak to nejde zapnout jen pro php. Takže nejde vytvořit soubor bez `\n` na konci.
Takže možnost je vytvářet `expected-stderr` v něčem jiném to je asi nejrozumnější a vyhodit ten trim, nebo escapovat a pak převádět '\n' na "\n" ale to je zas problém co když v stderr má být '\n' :D

Reply:
myslim, ze by se to nejlip resil tim kdyby tady https://github.com/keboola/datadir-tests/blob/35d71a04db47086c66a310d646a1156f13323709/src/AbstractDatadirTestCase.php#L150 nebyl `assertSame`, ale `assertContains`

Reply:
Toť otázka čekam bych právě spíš exact match. Teď pushnul změnu vyhodil ten trim a aktualizoval test a přidal readme

Reply:
exact match na stdout a stderr je dost nepraktickej, se tam objevi nejakej timstampt nebo idcko a ses v pytli

Reply:
Jako ten `assertContains` dává smysl, akorát by to teda byl asi BC break.

Reply:
Minimálně co by šlo je dát trim na obě místa. To je minizměna. Zbytek můžem řešit, až to nastane.

Reply:
(tj. trim vstupu ze souboru a trim i toho výstupu)

Reply:
to, ze je exact match na stderr a stdout neprakticky uz davno nastalo :)

Reply:
A contains je zas nepraktický v tom, že není jisté jestli tam není něco zajímavě rozbité "navíc"...

Reply:
Zrovna u testů je to taková nepříjemná vlastnost. 

Ale možná by šlo, že by to po řádcích porovnávalo přes contains nebo možná líp přes regex a když by to nematchnulo, tak by to failnulo... Co?

Reply:
nekomplikoval bych to - ja na ten stderr hledim jak na `expectExceptionMessage` a tam pisu "File not found" a ne "AWS:S3Clienterrorblbalbahovnohovnofailedtowhatverreason: File not found", protoze jsou jen dve veci co muzou nastat - bud se ta chyba nepropaguje, coz resi match. Nebo se propaguje blbe - app<>user - a to resi exit kod.

Reply:
A úplně ze stejného důvodu je špatně i moc krátký `expectExceptionMessage`. Někdo může spoléhat na to, že `if (strpos('AWS:S3Clienterror', $e->getMessage()) === 0) ...`

Ale jo. Holt tohle musíme hlídat v review no...

Reply:
Změnil sem to na `assertContains`